presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -153,4 +215,77 @@ static long timeUnitsToMillis(String timeWithUnits)
     {
         return Duration.valueOf(timeWithUnits).toMillis();
     }
+
+    private static class EffectiveLimits
+    {
+        private ZonedDateTime now;
+        private String maxQueryRangeDurationStr;
+        private Optional<PrometheusPredicateTimeInfo> maybePredicateRange;
+        private ZonedDateTime nowOrUpperBound;
+        private java.time.Duration maxQueryRangeDuration;
+
+        public EffectiveLimits(LocalDateTime now, String maxQueryRangeDurationStr, Optional<PrometheusPredicateTimeInfo> maybePredicateRange)
+        {
+            this.now = ZonedDateTime.of(now, ZoneId.systemDefault());
+            this.maxQueryRangeDurationStr = maxQueryRangeDurationStr;
+            this.maybePredicateRange = maybePredicateRange;
+        }
+
+        public ZonedDateTime getNowOrUpperBound()
+        {
+            return nowOrUpperBound;
+        }
+
+        public java.time.Duration getMaxQueryRangeDuration()
```

Comment:
Maybe we should call this `getLowerBound` and `maxQueryRangeDuration` (currently in invoke) to lowerBound. I think we're using names representing about how we're using this EffectiveLimits class into the implementation when, if we look at it separately it would be generally about upper/lower bounds.

Reply:
The code outside EffectiveLimits is building the chunked query to Prom. For the instant query we're using with Prom, the overall query needs an upperBound (time) and duration. The predicate defines with the idea of upper and lower, but EffectiveLimits conditionally rewrites the upperBound and duration as needed by the calling code:
```java
        EffectiveLimits effectiveLimits = new EffectiveLimits(defaultUpperBound, maxQueryRangeDurationStr, predicateRange);
        ZonedDateTime upperBound = effectiveLimits.getUpperBound();
        java.time.Duration maxQueryRangeDuration = effectiveLimits.getMaxQueryRangeDuration();
```

Reply:
Right, I get that, I mean't simply naming it differently. So it would still be used the same way, but like this:
```
        EffectiveLimits effectiveLimits = new EffectiveLimits(defaultUpperBound, maxQueryRangeDurationStr, predicateRange);
        ZonedDateTime upperBound = effectiveLimits.getUpperBound();
        java.time.Duration maxQueryRangeDuration = effectiveLimits.getLowerBound();
```

Just stuck out to me that EffectiveLiimits is more generically about bounded time ranges and is actually semi-agnostic of Prometheus but is using terminology specific to Prometheus like max query range. It's not a big deal, I just thought maybe we could use consistent terminology within EffectiveLimits and and then call it with our particular use case in mind.

Reply:
Actually I kinda see how this makes sense as it is now that I found how PrometheusPredicateTimeInfo is used.